sources/packages/api/src/route-controllers/restriction/restriction.controller.ts

```diff
@@ -35,10 +67,192 @@ export class RestrictionController extends BaseController {
     return studentRestrictions?.map((studentRestriction) => ({
       restrictionId: studentRestriction.id,
       restrictionType: studentRestriction.restriction.restrictionType,
+      restrictionCategory: studentRestriction.restriction.restrictionCategory,
       description: studentRestriction.restriction.description,
       createdAt: studentRestriction.createdAt,
       updatedAt: studentRestriction.updatedAt,
       isActive: studentRestriction.isActive,
     }));
   }
+  /**
+   * REST API to provide the list of restriction categories for drop-down.
+   * @returns Categories option list.
+   */
+  @Groups(UserGroups.AESTUser)
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Get("/categories/options-list")
+  async getCategoriesOptionsList(): Promise<OptionItem[]> {
+    const categories =
+      await this.restrictionService.getAllRestrictionCategories();
+    return categories.map((category) => ({
+      id: category.id,
+      description: category.restrictionCategory,
+    }));
+  }
+
+  /**
+   * REST API to provide the list of restriction reasons for selected category.
+   * @returns Reasons option list.
```

Comment:
pls add comment for `restrictionCategory` params

Reply:
Addressed.